pinot-core/src/main/java/org/apache/pinot/core/common/DataSource.java

```diff
@@ -45,4 +46,7 @@
    * Returns the bloom filter for the data source if exists, or {@code null} if not.
    */
   public abstract BloomFilterReader getBloomFilter();
+
+  public abstract PresenceVectorReader getPresenceVector();
```

Comment:
This is more like an NullValueVector

Reply:
PresenceVectorReader has a method called "isPresent(docId)" . So it is in fact a presence value vector (although internally it uses a null vector).

Reply:
Modified to NullValueVector